cmd/podman/shared/create.go

```diff
@@ -322,7 +322,7 @@ func configurePod(c *GenericCLIResults, runtime *libpod.Runtime, namespaces map[
 	if (namespaces["uts"] == cc.Pod) || (!c.IsSet("uts") && pod.SharesUTS()) {
 		namespaces["uts"] = fmt.Sprintf("container:%s", podInfraID)
 	}
-	return namespaces, nil
+	return namespaces, pod.ID(), nil
```

Comment:
This should be either of:
- infra container's ID (only then `/var/lib/containers/storage/overlay-containers/${INFRA_CONTAINER_ID}/userdata/config.json` exists for other containers in the same pod to latch onto)
- if we don't have an infra container, the ID of whichever container is running with ContainerType == `sandbox` (there should always be only one sandbox container within a pod, but the same condition applies)

Either of these containers has to be running in order to be able to attach other containers with ContainerType == `container` into the sandbox.

Reply:
From the sounds of it @zer0def, it seems like we should only allow this kind of sharing if the sandbox container is an infra container. There is already mechanisms in place to guarentee the infra container isn't deleted before other containers in the pod, which seems like something that could effect kata.

For instance, if the first pod in the container is created, another is added (and attached to its vm) and the first is deleted, I'm pretty sure that container's config.json is going to be removed. Would that negatively effect the other containers in the pod?

Reply:
Indeed, the intended idea would be to have the infra container be the pod's sandbox container, but I wouldn't immediately dismiss other ideas, if there's sufficient motivation for doing so with careful execution.

Reply:
okay, I'll stick to an infra container for now